kedro/io/partitioned_dataset.py

```diff
@@ -50,19 +54,54 @@ class PartitionedDataSet(AbstractDataSet):
         >>>       index: False
         >>>   path: data/04_cv/
         >>>   filename_suffix: ".csv"
-        >>>
-        >>>   downloaded_data: # example with data available in multiple partitions
-        >>>     type: PartitionedDataSet
-        >>>     path: demo/01_raw/downloaded_station_data
-        >>>     dataset:
-        >>>       type: pandas.CSVDataSet
-        >>>       load_args:
-        >>>         sep: ','
-        >>>         index_col: 0
-        >>>     filename_suffix: '.csv'
+
+        >>> downloaded_data: # example with data available in multiple partitions
+        >>>   type: PartitionedDataSet
+        >>>   path: demo/01_raw/downloaded_station_data
+        >>>   dataset:
+        >>>     type: pandas.CSVDataSet
+        >>>     load_args:
+        >>>       sep: ','
+        >>>       index_col: 0
+        >>>   filename_suffix: '.csv'
 
 
     Example using Python API:
+    ::
+        >>> import pandas as pd
+        >>> from kedro.io import PartitionedDataSet
+
+        # Create a fake pandas dataframe with 10 rows of data
+        >>> df = pd.DataFrame([{"DAY_OF_MONTH": str(i), "VALUE": i} for i in range(1, 11)])
+
+        # Convert it to a dict of pd.DataFrame with DAY_OF_MONTH as the dict key
+        >>> dict_df = dict(tuple(df.groupby("DAY_OF_MONTH")))
```

Comment:
@AntonyMilneQB As the comment suggests, it's a Dict[str, pd.Dataframe]

Reply:
Ah, I see. Might be worth just showing what `dict_df` looks like in a comment:
```
# Convert it to a dict of pd.DataFrame with DAY_OF_MONTH as the dict key.
# e.g. dict_df["1"] is pd.DataFrame({"DAY_OF_MONTH": "1", "VALUE": 1}, index=[0])
```

Or maybe doing this a bit more explicitly as something like this?
```
dict_df = {row["DAY_OF_MONTH"]: row for index, row in df.iterrows()}
```

Or even going through `df.iloc[...]`.

Reply:
@AntonyMilneQB Yeah I agree, this is one of those pandas tricks that isn't very explicit but are necessary for performance. For documentation purposes it could be more explicit

```python
dict_df = {day_of_month: df[df["DAY_OF_MONTH"] == day_of_month]
     for day_of_month in df['DAY_OF_MONTH']}
```